EIPS/eip-7701.md

```diff
@@ -40,265 +43,71 @@ Transactions of this type are referred to as "AA transactions".
-| 0xF3         | `execution_gas_cost`             | 32        |              | only `role_paymaster_post_op`             |
-
-### Limitations on `TXPARAM*` opcodes
-
-The `TXPARAM*` opcodes are only enabled in frames with a correct `current_frame_role`.
-Calling these opcodes in another context, or with invalid `txparam_id`, returns zero values and zero lengths.
-
-| `current_frame_role`                                                            | allowed `txparam_id`                                                                                       |
-|---------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------|
-| 0 (none)                                                                        | None, `TXPARAM*` opcodes always return zero value and zero length                                          |
-| `role_sender_deployment`, `role_sender_validation`, `role_paymaster_validation` | All tx parameters, `current_frame_role`, `tx_hash_for_signature`                                           |
-| `role_paymaster_post_op`                                                        | All tx parameters, `current_frame_role`, `tx_hash_for_signature`, `execution_status`, `execution_gas_cost` |
-| `role_sender_execution`                                                         | Only `current_frame_role`                                                                                  |
-
-
-Contact should first call `TXPARAMLOAD 0xF0` (`current_frame_role`) to determine the current frame role.
-
-In case `current_frame_role` is not set for the current frame it has a default value of `0`.
-
-### `TXPARAM*` opcodes gas prices
-
-The new opcodes have gas prices equivalent to corresponding `CALLDATA*` opcodes:
-
-1. `TXPARAMLOAD` - static value of `3` gas
-2. `TXPARAMSIZE`- static value of `2` gas
-3. `TXPARAMCOPY` - dynamic formula
-
-    ```
-    minimum_word_size = (size + 31) / 32
-    static_gas = 3
-    dynamic_gas = 3 * minimum_word_size + memory_expansion_cost
-    ```
-
-#### Time range validity (optional)
-
-If the `valid_until` field is non-zero, the transaction is only valid for inclusion in a block with a timestamp at most `valid_until` value.
-Similarly, the transaction is only valid for inclusion in blocks with a timestamp at most the `valid_after` value.
-
-#### Sender Deployment (optional)
-
-Inputs to the `deployer` contract are not defined by the protocol and are controlled by the `deployer_data` parameter.
-
-The sender deployment frame MUST result in the `sender` address becoming initialized with contract code.
-
-This step is performed with the `role_sender_deployment` role.
+### `CURRENT_ROLE` and `ACCEPT_ROLE` opcodes
 
-#### Sender Validation
+`current_context_role` is a context variable set by the AA transaction to the current role.
```

Comment:
```suggestion
`current_frame_role` is a context variable set by the AA transaction to the current role.
```

Reply:
@drortirosh Didn't you just rename it in the other direction? Which way do you like it? Why? Please finalize the decision. Does it even matter that much?